### ISSUE TITLE
superset behavior is not same as set

### ISSUE THREAD

**Issue by handloomweaver** [simonpercivall/orderedset#8]
There seems to be a bug around subset/superset. If using 2 orderedsets it does not behave as 2 sets do. If using 1 orderedset and an ordinary set it behaves as you'd expect. I've written a lot of code that uses this library and I'm about to push into production. Can you get this fixed asap? Please

     #Ordinary Set subset:
     In [33]: a = set()
     In [34]: b = set()
     In [35]: a.add(1)
     In [36]: a.add(2)
     In [37]: a.add(3)
     In [38]: b.add(2)
     In [39]: b.add(3)
     In [40]: b <= a
     Out[40]: True

     #OrderedSet to OrderedSet subset:
     In [41]: a = OrderedSet()
     In [42]: b = OrderedSet()
     In [43]: a.add(1)
     In [44]: a.add(2)
     In [45]: a.add(3)
     In [46]: b.add(2)
     In [47]: b.add(3)
     In [48]: b <= a
     Out[48]: False

     #OrderedSet to set subset:
     In [3]: a = OrderedSet()
     In [4]: a.add(1)
     In [5]: a.add(2)
     In [6]: a.add(3)
     In [7]: b = set()
     In [8]: b.add(2)
     In [9]: b.add(3)
     In [10]: b <= a
     Out[10]: True

**Comment by simonpercivall** [comment]
So the difference is this: An OrderedSet compared to another OrderedSet cares about order, which means that comparisons against other ordered collections _also_ cares about order. So in this regard, an OrderedSet compares more like a list than a set, when using the "less than or equal" operator. (Of course, it is wrong that the `issubset` method, which is implemented by using that operator, also behaves like a list comparison.)

TLDR: I'm not entirely sure that it wouldn't _also_ be a bug to make comparisons behave more like set comparisons than list comparisons.

**Comment by handloomweaver** [comment]
I get what you're saying, an OrderedSet is conceptually somewhere between a list and a set. 
But first and foremost these are sets which are a mathematical concept as you know - containers of unique values. Subset, Superset relate to comparison between containers of unique values. They methods on the set interface must be maintained otherwise these are not sets, ordered or not. 
Also it is extremely confusing API design to have different behaviour for the "less than or equal" operator. At the very very least, the explicit methods like .issubset must work this way. These are sets. If you disagree, then call them something else. No?

However, it is surely not a bug to maintain all of set()'s interfa
